api/query/query.go

```diff
@@ -61,6 +62,32 @@ func (qh queryHandler) processInput(w http.ResponseWriter, r *http.Request) (*sh
 	return &filters, testRuns, summaries, nil
 }
 
+func (qh queryHandler) validateSummaryVersions(v url.Values) (bool, error) {
+	filters, err := shared.ParseQueryFilterParams(v)
+	if err != nil {
+		return false, err
+	}
+	testRuns, filters, err := qh.getRunsAndFilters(filters)
+	if err != nil {
+		return false, err
+	}
+
+	for _, testRun := range testRuns {
+		summaryURL := shared.GetResultsURL(testRun, "")
+		// All summary URLs end with "-summary.json.gz". We remove this and
+		// add the correct suffix for the version file.
+		const suffixOffset = 16
+		url := summaryURL[:len(summaryURL)-suffixOffset] + "_version.txt"
+		mkey := getRedisVersionKey(testRun)
+		var data []byte
+		err := qh.dataSource.Get(mkey, url, &data)
+		if len(data) == 0 || err != nil {
```

Comment:
Since `len(data) == 0` is a condition where `validateSummaryVersions` needs to return, you should return an error. 
This function returns either: `false, err` or `true, nil`. By returning an error for this case, you can simplify the return type to just `error` instead of `(bool, error)`. (A bad alternative would be changing the name of this function to `areSummaryVersionsValidated` and return just a bool. But you would need to do a lot of logging since removing the error loses a lot of useful data.)

An example of splitting that out:
```go
if err !=nil {
    return err
} else if len(data) == 0 {
    return fmt.Errorf("custom error message: %s", mkey)
}
```

Callers of `validateSummaryVersions` can have a simple expectation that an error means something happened unexpectedly if an error is returned. If the caller needs to differentiate how to proceed after getting an error from `validateSummaryVersions`, it can always compare the errors.

Reply:
I see, that's true. I'll add an accurate error to this case.